lib/valid_url.rb

```diff
@@ -26,55 +25,44 @@ def validate_each(record, attribute, value)
       protected
 
       # add common protocol by default
-      def ensure_protocol url
-        if url[/\A(http|https):\/\//i]
+      def ensure_protocol(url)
+        if url[/\A(#{PROTOCOLS.join('|')}):\/\//i]
           url
         else
           "http://" + url
         end
       end
 
       # http and https are accepted
-      def valid_scheme? scheme
+      def valid_scheme?(scheme)
         UrlValidator::PROTOCOLS.include?(scheme.mb_chars.downcase.to_s)
       end
 
-      def valid_host? host
+      def valid_host?(host)
         return false unless host.present? && valid_characters?(host)
         labels = host.split('.')
-        valid_length?(host) && valid_labels?(labels) && (valid_ip?(host) || valid_zone?(labels.last))
+        valid_length?(host) && valid_labels?(labels)
       end
 
       # each label must be between 1 and 63 characters long
-      def valid_labels? labels
+      def valid_labels?(labels)
         labels.count >= 2 && labels.all?{ |label| label.length >= 1 && label.length <= 63 }
       end
 
       # entire hostname has a maximum of 253 characters
-      def valid_length? host
+      def valid_length?(host)
         host.length <= 253
       end
 
-      # only existent domain name zones
-      def valid_zone? zone
-        UrlValidator::ZONES.include?(zone.mb_chars.downcase.to_s)
```

Comment:
I don't think I like this change after all :/ Because what if someone by mistakes enters `my-domain.con` instead of `my-domain.com`? I think the least we can do is check if the top level domain is correct because there is a publicly available list hosted by IANA.